internal/state/driver_volumes.go

```diff
@@ -159,13 +159,24 @@ func (s *stateDriver) getDriverAdditionalConfigs(ctx context.Context, cr *v1alph
 			if err != nil {
 				return nil, fmt.Errorf("ERROR: failed to get destination directory for custom repo config: %w", err)
 			}
-			volumeMounts, itemsToInclude, err := s.createConfigMapVolumeMounts(ctx, s.namespace,
-				cr.Spec.CertConfig.Name, destinationDir)
-			if err != nil {
-				return nil, fmt.Errorf("ERROR: failed to create ConfigMap VolumeMounts for custom certs: %w", err)
+
+			if cr.Spec.LicensingConfig.SecretName != "" {
+				volumeMounts, itemsToInclude, err := s.createSecretVolumeMounts(ctx, s.namespace,
+					cr.Spec.CertConfig.Name, destinationDir)
```

Comment:
I don't think we should be referencing `cr.Spec.CertConfig` here
```suggestion
					cr.Spec.LicensingConfig.SecretName, destinationDir)
```

Reply:
// set any custom ssl key/certificate configuration provided
if cr.Spec.IsCertConfigEnabled() { 

we are passing cert name for custorm certificate configuration . 
Please check again @tariq1890